File: SDK/DataModel/StaticData/HeroType.cs
using Newtonsoft.Json;
using System.Linq;

namespace Raid.DataModel
{
    public class HeroType
    {
        [JsonProperty("typeId")]
        public int TypeId;

        [JsonProperty("name")]
        public LocalizedText Name;

        [JsonProperty("affinity")]
        public string Affinity;

        [JsonProperty("faction")]
        public string Faction;

        [JsonProperty("role")]
        public string Role;

        [JsonProperty("rarity")]
        public string Rarity;

        [JsonProperty("ascended")]
        public int Ascended;

        [JsonProperty("modelName")]
        public string ModelName;

        [JsonProperty("avatarKey")]
        public string AvatarKey;

        [JsonProperty("leaderSkill")]
        public LeaderStatBonus LeaderSkill;

        [JsonProperty("skillTypeIds")]
        public int[] SkillTypeIds;

        [JsonProperty("unscaledStats")]
        public Stats UnscaledStats;

        [JsonProperty("brain")]
        public string Brain;
    }
}

File: SDK/WebClient/src/Types/HeroType.ts
import { Affinity, Faction, HeroRole, Rarity } from './Enums';

export interface LocalizedText {
  key: string;
  defaultValue: string;
  localizedValue: string;
}

export interface Stats {
  Health: number;
  Attack: number;
  Defense: number;
  Speed: number;
  Resistance: number;
  Accuracy: number;
  CriticalChance: number;
  CriticalDamage: number;
  CriticalHeal: number;
}

export interface StatBonus {
  kind: string;
  absolute: boolean;
  value: number;
}

export interface LeaderStatBonus extends StatBonus {
  areaTypeId: string;
  affinity: Affinity;
}

export interface HeroType {
  typeId: number;
  name: LocalizedText;
  affinity: Affinity;
  faction: Faction;
  role: HeroRole;
  rarity: Rarity;
  ascended: number;
  modelName: string;
  avatarKey: string;
  leaderSkill: LeaderStatBonus;
  skillTypeIds: number[];
  unscaledStats: Stats;
  brain: string;
}
